src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1470,7 +1470,36 @@ internal async Task HandleUpdateAgentNotebookRunPinRequest(UpdateAgentNotebookRu
             });
         }
 
-        public AgentNotebookHistoryResult GetAgentNotebookHistories(
+        internal async Task HandleDeleteNotebookMaterializedRequest(DeleteNotebookMaterializedParams parameters, RequestContext<ResultStatus> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                var result = new ResultStatus();
+                try
+                {
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                                                parameters.OwnerUri,
+                                                out connInfo);
+                    // Calling update helper function
+                    await AgentNotebookHelper.DeleteMaterializedNotebook(
+                        connInfo,
+                        parameters.agentNotebookHistory,
+                        parameters.TargetDatabase);
+                    result.Success = true;
+                }
+                catch (Exception e)
+                {
+                    result.Success = false;
+                    result.ErrorMessage = e.ToString();
+
```

Comment:
Normally you'd just do await result.SendError(e);

Reply:
Make an issue to make these exceptions call SendError instead

---
In reply to: [321924046](https://github.com/microsoft/sqltoolsservice/pull/862#discussion_r321924046) [](ancestors = 321924046)